pkg/memlimit/memlimit.go

```diff
@@ -0,0 +1,41 @@
+package memlimit
+
+import (
+	"os"
+	"runtime/debug"
+	"strconv"
+
+	"github.com/stackrox/rox/pkg/logging"
+)
+
+var log = logging.LoggerForModule()
+
+func SetMemoryLimit() {
+	var limit int64
+
+	defer func() {
+		if limit == 0 {
+			log.Warn("Memory limit left unset")
+			return
+		}
+
+		limitGi := float64(limit) / 1024 / 1024 / 1024
+		log.Infof("Memory limit set: %.2fGi", limitGi)
+	}()
+
+	var err error
+	if goLimit := os.Getenv(`GOMEMLIMIT`); goLimit != "" {
+		if limit, err = strconv.ParseInt(goLimit, 10, 64); err == nil {
```

Comment:
> GOMEMLIMIT is a numeric value in bytes with an optional unit suffix. The supported suffixes include B, KiB, MiB, GiB, and TiB. 

https://pkg.go.dev/runtime#:~:text=the%20Go%20program.-,GOMEMLIMIT,-is%20a%20numeric